app/models/post_action_type.rb

```diff
@@ -90,40 +119,40 @@ def notify_flag_type_ids
 
     def notify_flag_types
       return flag_settings.notify_types if overridden_by_plugin_or_skipped_db?
-      flag_enum(all_flags.select(&:notify_type))
+      flag_enum(all_flags.select { |flag| flag[:notify_type] })
     end
 
     def topic_flag_types
       if overridden_by_plugin_or_skipped_db?
         flag_settings.topic_flag_types
       else
-        flag_enum(all_flags.select { |flag| flag.applies_to?("Topic") })
+        flag_enum(all_flags.select { |flag| flag[:applies_to].include?("Topic") })
       end
     end
 
     def disabled_flag_types
-      flag_enum(all_flags.reject(&:enabled))
+      flag_enum(all_flags.reject { |flag| flag[:enabled] })
     end
 
     def enabled_flag_types
-      flag_enum(all_flags.filter(&:enabled))
+      flag_enum(all_flags.filter { |flag| flag[:enabled] })
     end
 
     def additional_message_types
       return flag_settings.with_additional_message if overridden_by_plugin_or_skipped_db?
-      flag_enum(all_flags.select(&:require_message))
+      flag_enum(all_flags.select { |flag| flag[:require_message] })
     end
 
     def names
-      all_flags.pluck(:id, :name).to_h
+      all_flags.map { |f| [f[:id], f[:name]] }.to_h
```

Comment:
```suggestion
      all_flags.reduce({}) { |acc, f| acc[f[:id]] = f[:name]; acc }
```

Generating an array just to convert it to a hash feels abit wasteful here. Perhaps we can just use `Array#reduce`?